packages/core/src/geometry/BufferSystem.ts

```diff
@@ -0,0 +1,221 @@
+ * - Transform feedback information. (WebGL 2 only feature)
+ *
+ * This system will handle the binding of buffers to the GPU as well as uploading
+ * them. With this system, you never need to work directly with GPU buffers, but instead work with
+ * the PIXI.Buffer class.
+ *
+ *
+ * @class
+ * @memberof PIXI
+ */
+export class BufferSystem implements ISystem
+{
+    CONTEXT_UID: number;
+    gl: IRenderingContext;
+
+    /** Cache for all buffers by id, used in case renderer gets destroyed or for profiling */
+    readonly managedBuffers: {[key: number]: Buffer};
+
+    /** Cache keeping track of the base bound buffer bases */
+    readonly boundBufferBases: {[key: number]: Buffer};
+
+    private renderer: Renderer;
+
+    /**
+     * @param {PIXI.Renderer} renderer - The renderer this System works for.
+     */
+    constructor(renderer: Renderer)
+    {
+        this.renderer = renderer;
+        this.managedBuffers = {};
+        this.boundBufferBases = {};
+    }
+
+    /**
+     * @ignore
+     */
+    destroy(): void
+    {
+        this.renderer = null;
+    }
+
+    /**
+     * Sets up the renderer context and necessary buffers.
+     */
+    protected contextChange(): void
+    {
+        this.disposeAll(true);
+
+        this.gl = this.renderer.gl;
+
+        // TODO fill out...
+        this.CONTEXT_UID = this.renderer.CONTEXT_UID;
+    }
+
+    /**
+     * this binds specified buffer. OOn first run, it will create the webGL buffers for the context too
+     *
+     * @param buffer - the buffer to bind to the renderer
+     */
+    bind(buffer: Buffer): void
+    {
+        const { gl, CONTEXT_UID } = this;
+
+        const glBuffer = buffer._glBuffers[CONTEXT_UID] || this.createGLBuffer(buffer);
+
+        gl.bindBuffer(buffer.type, glBuffer.buffer);
+    }
+
+    /**
+     * binds a buffer to a base. A cache is used so a buffer will not be bound again if already bound.
+     * Only used by the uniform buffers
+     *
+     * @param buffer - the buffer to bind
+     * @param index - the base index to bind it to.
+     */
+    bindBufferBase(buffer: Buffer, index: number): void
+    {
+        const { gl, CONTEXT_UID } = this;
+
+        if (this.boundBufferBases[index] !== buffer)
+        {
+            const glBuffer = buffer._glBuffers[CONTEXT_UID] || this.createGLBuffer(buffer);
+
+            this.boundBufferBases[index] = buffer;
+
+            gl.bindBufferBase(gl.UNIFORM_BUFFER, index, glBuffer.buffer);
+        }
+    }
+
+    /**
+     * Binds a buffer whilst also binding its range.
+     * This will make the buffer start from the offset supplied rather than 0 when it is read.
+     *
+     * @param buffer - the buffer to bind
+     * @param offset - the offset to bind at (this is blocks of 256). 0 = 0, 1 = 256, 2 = 512 etc
+     * @param index - the base index to bind at, defaults to 0
+     */
+    bindBufferRange(buffer: Buffer, offset?: number, index?: number): void
```

Comment:
I'm concerned about the order of the parameters `offset, index`. It would be more intuitive for `index` to come first to be consistent with `bindBufferBase`'s pattern.

Also, lets separate the doc brief (add a line between first & second sentence)

Reply:
agree! will flip em!